tests/rados/test_four_node_ecpool.py

```diff
@@ -0,0 +1,218 @@
+    pool_obj = PoolFunctions(node=cephadm)
+
+    if not mon_obj.verify_set_config(
+        section="mon", name="mon_osd_down_out_subtree_limit", value="host"
+    ):
+        log.error(
+            "mon_osd_down_out_subtree_limit not set on the cluster. setting the same now"
+        )
+        mon_obj.set_config(
+            section="mon", name="mon_osd_down_out_subtree_limit", value="host"
+        )
+
+    if not mon_obj.verify_set_config(
+        section="osd", name="osd_async_recovery_min_cost", value="1099511627776"
+    ):
+        log.error(
+            "osd_async_recovery_min_cost not set on the cluster. setting the same now"
+        )
+        mon_obj.set_config(
+            section="osd", name="osd_async_recovery_min_cost", value="1099511627776"
+        )
+
+    # Creating the EC pool
+    ec_config = config.get("ec_pool")
+    pool_name = ec_config["pool_name"]
+    if not rados_obj.create_erasure_pool(name=ec_config["profile_name"], **ec_config):
+        log.error("Failed to create the EC Pool")
+        return 1
+
+    cmd = "ceph osd pool autoscale-status"
+    pool_status = rados_obj.run_ceph_command(cmd=cmd)
+
+    for entry in pool_status:
+        if entry["pool_name"] == pool_name:
+            if entry["pg_autoscale_mode"] == "off":
+                log.error(
+                    f"Pg autoscaler turned off for the new pool : {entry['pool_name']} "
+                )
+                return 1
+
+    init_pg_count = rados_obj.get_pool_property(pool=pool_name, props="pg_num")[
+        "pg_num"
+    ]
+    log.debug(f"init PG count on the pool : {init_pg_count}")
+
+    bulk = pool_obj.get_bulk_details(pool_name=pool_name)
+    if bulk:
+        log.error("Expected bulk flag should be False.")
+        raise Exception("Expected bulk flag should be False.")
+
+    log.debug(f"Bulk flag not enabled on pool {pool_name}, Proceeding to enable bulk")
+
+    new_bulk = pool_obj.set_bulk_flag(pool_name=pool_name)
+    if not new_bulk:
+        log.error("Expected bulk flag should be True.")
+        raise Exception("Expected bulk flag should be True.")
+    # Sleeping for 60 seconds for bulk flag application and PG count to be increased.
+    time.sleep(60)
+
+    pg_count_bulk_true = rados_obj.get_pool_details(pool=pool_name)["pg_num_target"]
+    log.debug(
+        f"PG count on pool {pool_name} post addition of bulk flag : {pg_count_bulk_true}"
+        f"Starting to wait for PG count on the pool to go from {init_pg_count} to"
+        f" {pg_count_bulk_true} while checking for PG inactivity"
+    )
+
+    endtime = datetime.datetime.now() + datetime.timedelta(seconds=10000)
+    while datetime.datetime.now() < endtime:
+        pool_pg_num = rados_obj.get_pool_property(pool=pool_name, props="pg_num")[
+            "pg_num"
+        ]
+        if pool_pg_num == pg_count_bulk_true:
+            log.debug(f"PG count on pool {pool_name} is achieved")
+            break
+        log.debug(
```

Comment:
Please consider changing this to log.info for new users to understand where the workflow is waiting to complete

Reply:
Done